sources/engine/Stride.UI/Rendering/UI/UIRenderFeature.cs

```diff
@@ -54,29 +47,23 @@ protected override void InitializeCore()
 
             Name = "UIComponentRenderer";
             game = RenderSystem.Services.GetService<IGame>();
-            input = RenderSystem.Services.GetService<InputManager>();
             uiSystem = RenderSystem.Services.GetService<UISystem>();
-            graphicsDeviceService = RenderSystem.Services.GetSafeServiceAs<IGraphicsDeviceService>();
 
+            // TODO: Change Batch to be local to the UIRenderFeature and remove dep on UISystem. Requires refactoring ElementRenderer.
```

Comment:
I looked in to it, but would require a pretty non-trivial refactor, and the current rendering system has signifigant limitations. I am finishing up a proposal for a redesign of how the look of UIElements is handled, which would replace the current system.

So due to the amount of work involved in the refactor, and the pending rewrite(hopefully). It didn't seem worth it to invest the time right now. The main reason I left the TODO is so that if I for some reason, don't get around to the redesign (or it is rejected). Somone else can understand why there is a circular dependency between `UISystem` and `UIRenderFeature`. Not sure if this is the correct way to handle it?